Core/clim-basic/windowing/ports.lisp

```diff
@@ -68,8 +69,9 @@
       (when (null port-class)
         (error "No CLIM backends have been loaded!"))
       (or (find server-path *all-ports* :test #'equal :key #'port-server-path)
-          (first (push (make-instance port-class :server-path server-path)
-                       *all-ports*))))))
+          (with-lock-held (*all-ports-lock*)
+            (first (push (make-instance port-class :server-path server-path)
+                         *all-ports*)))))))
```

Comment:
The critical section starts on the previous line: two threads could perform the `find`, both find nothing, then both (safely but incorrectly) add a new port.

Reply:
right, fixed